these are introduced with the 1.91.0 release and could be investigated further,
if there is ground for changing our build process

--- a/src/bootstrap/src/core/sanity.rs
+++ b/src/bootstrap/src/core/sanity.rs
@@ -345,15 +345,6 @@ than building it.
                 || build.config.rust_std_features.contains("compiler-builtins-c"))
         {
             let cc_tool = build.cc_tool(*target);
-            if !cc_tool.is_like_clang() && !cc_tool.path().ends_with("emcc") {
-                // emcc works as well
-                panic!(
-                    "Clang is required to build C code for Wasm targets, got `{}` instead\n\
-                    this is because compiler-builtins is configured to build C source. Either \
-                    ensure Clang is used, or adjust this configuration.",
-                    cc_tool.path().display()
-                );
-            }
         }
 
         if (target.contains("-none-") || target.contains("nvptx"))
@@ -406,16 +397,6 @@ $ pacman -R cmake && pacman -S mingw-w64
                 );
             }
         }
-
-        // For testing `wasm32-wasip2`-and-beyond it's required to have
-        // `wasm-component-ld`. This is enabled by default via `tool_enabled`
-        // but if it's disabled then double-check it's present on the system.
-        if target.contains("wasip")
-            && !target.contains("wasip1")
-            && !build.tool_enabled("wasm-component-ld")
-        {
-            cmd_finder.must_have("wasm-component-ld");
-        }
     }
 
     if let Some(ref s) = build.config.ccache {
